media/js/plugincheck/check-plugins-redesign.js

```diff
@@ -0,0 +1,327 @@
+    var readerRegEx = /Adobe \b(Reader|Acrobat)\b.*/;
+    var iconFor = function (pluginName) {
+        if (pluginName.indexOf('Flash') >= 0) {
+            return 'icon-flash.png';
+        } else if (pluginName.indexOf('Java') >= 0) {
+            return 'icon-java.png';
+        } else if (pluginName.indexOf('QuickTime') >= 0) {
+            return 'icon-quicktime.png';
+        } else if (pluginName.indexOf('DivX') >= 0) {
+            return 'icon-divx.png';
+        } else if (pluginName.indexOf('Totem') >= 0) {
+            return 'icon-totem.png';
+        } else if (pluginName.indexOf('Flip4Mac') >= 0) {
+            return 'icon-flip4mac.png';
+        } else if (pluginName.indexOf('WindowsMediaPlayer') >= 0) {
+            return 'icon-wmp.png';
+        } else if (pluginName.indexOf('VLC') >= 0) {
+            return 'icon-vlc.png';
+        } else if (pluginName.indexOf('Silverlight') >= 0) {
+            return 'icon-silverlight.png';
+        } else if (pluginName.indexOf('Shockwave') >= 0) {
+            return 'icon-shockwave.png';
+        } else if (pluginName.indexOf('RealPlayer') >= 0) {
+            return 'icon-real.png';
+        } else if (readerRegEx.test(pluginName)) {
+            return 'icon-acrobat.png';
+        } else if (pluginName.indexOf('Office Live') >= 0) {
+            return 'icon-officelive.png';
+        } else if (pluginName.indexOf('iPhoto') >= 0) {
+            return 'icon-iphoto.png';
+        } else {
+            return 'default.png';
+        }
+    };
+
+    /**
+     * Handles click events triggered from various plugin status buttons. Once the
+     * click event is received, it sends the relevant data about the interaction to GA.
+     */
+    function handleButtonInteractions() {
+        $pluginsContainer.on('click', 'a.action-link', function(event) {
+            window.dataLayer.push({
+                'event': 'plugincheck-interactions',
+                'interaction': 'button click',
+                'plugin-action': event.target.dataset['status'],
+                'plugin-name': event.target.dataset['name']
+            });
+        });
+    }
+
+    /**
+     * Renders a plugin object using associated template for each status.
+     * @param {object} plugin - Plugin object to render.
+     */
+    function showPlugin(plugin) {
+        var featuredPluginsSection = $('#sec-plugin-featured');
+        var featuredPluginsBody = $('#plugin-featured');
+        var featuredPluginsHtml = '';
+        var outdatedPluginsSection = $('#sec-plugin-outdated');
+        var outdatedPluginsBody = $('#plugin-outdated');
+        var outdatedPluginsHtml = '';
+        var unknownPluginsSection = $('#sec-plugin-unknown');
+        var unknownPluginsBody = $('#plugin-unknown');
+        var unknownPluginsHtml = '';
+        var upToDatePluginsSection = $('#sec-plugin-uptodate');
+        var upToDatePluginsBody = $('#plugin-uptodate');
+        var upToDatePluginsHtml = '';
```

Comment:
As this function is called multiple times, would it be better to declare the section & body vars outside (so they don't get re-declared on each call)?

Also, since there's only ever one chunk of HTML created per plugin, could we just use a single, general HTML container var instead of declaring one for each status?